### DISCUSSION TITLE
Globaleacks API documentation

### DISCUSSION THREAD

**Discussion by gianluca326** [globaleaks/globaleaks-whistleblowing-software#3617] [Q&A]
Vorrei sapere se per le API che sono definite nel codice qui:
https://github.com/globaleaks/GlobaLeaks/blob/main/backend/globaleaks/rest/api.py

vorrei sapere se i parametri per la request sono rimasti invariati per i vari metodi, o se sono variati (o aggiunti/rimossi).

Ci sarebbe la possibilità di avere uno swagger?

Grazie mille.

**Comment by evilaliv3** [comment]
Grazie per la domanda.

Rimossi o variati rispetto a che versione?

Le API di fatto sono spesso evolute in quanto il software vede un costante e atrtivo sviluppo.
Nel tempo stiamo cercando di standardizzarle e renderle documentate e disponibili perchè system integrator possano fruirne per farne integrazioni con alcuni software ma purtroppo non abbiamo al momento risorse dedicate a questa attività.

In merito agli swagger non ne disponiamo. E' stato ipotizzato di cercare di integrare una libreria o una riscrivere il motore REST del software per generarlo automaticamente ma ancora non abbiamo avanzamenti.

Voi siete esperti di questo dominio? Avete piacere / modo di prendere parte ad uno sviluppo in questo senso?

**Reply by gianluca326** [reply]
La versione che utilizzavamo è la 3.1.6
A questo punto non ci resta che verificare le API che utilizzavamo se funzionano ancora in riferimento al file di progetto api.py
Purtroppo non siamo esperti del dominio, e per adesso siamo anche noi in carenza di risorse, magari se riusciamo a preparare un minimo di documentazione cartacea la condivideremo sicuramente.
Grazie mille

**Comment by gianluca326** [comment]
Buongiorno,
stavo testando l'API

https://whistleblowing.parsec326.cloud/api/auth/authentication

passando il json:

{"username":"admin","password":"12345678","token":""}

ma mi restituisce :

{
   "error_message": "InternalServerError [Invalid request: No token and no session]",
   "error_code": 1,
   "arguments": ["Invalid request: No token and no session"]
}

potreste indicarmi come effettuare correttamente la request.
Grazie mille

**Reply by Bilox03** [reply]
Ciao, sei riuscito a risolvere? Ho lo stesso problema
